src/lirc.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022 Sean Young <sean@mess.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "defs.h"
+#include <linux/lirc.h>
+
+static const struct features {
+	int flag;
+	const char *name;
+} features[] = {
+	{ LIRC_CAN_REC_SCANCODE, "LIRC_CAN_REC_SCANCODE" },
+	{ LIRC_CAN_REC_MODE2, "LIRC_CAN_REC_MODE2" },
+	{ LIRC_CAN_GET_REC_RESOLUTION, "LIRC_CAN_GET_REC_RESOLUTION" },
+	{ LIRC_CAN_SEND_PULSE, "LIRC_CAN_SEND_PULSE" },
+	{ LIRC_CAN_SET_TRANSMITTER_MASK, "LIRC_CAN_SET_TRANSMITTER_MASK" },
+	{ LIRC_CAN_SET_SEND_CARRIER, "LIRC_CAN_SET_SEND_CARRIER" },
+	{ LIRC_CAN_SET_SEND_DUTY_CYCLE, "LIRC_CAN_SET_SEND_DUTY_CYCLE" },
+	{ LIRC_CAN_SET_REC_CARRIER, "LIRC_CAN_SET_REC_CARRIER" },
+	{ LIRC_CAN_SET_REC_CARRIER_RANGE, "LIRC_CAN_SET_REC_CARRIER_RANGE" },
+	{ LIRC_CAN_USE_WIDEBAND_RECEIVER, "LIRC_CAN_USE_WIDEBAND_RECEIVER" },
+	{ LIRC_CAN_MEASURE_CARRIER, "LIRC_CAN_MEASURE_CARRIER" },
+	{ LIRC_CAN_SET_REC_TIMEOUT, "LIRC_CAN_SET_REC_TIMEOUT" },
+};
```

Comment:
We replaced this approach with xlat files roughly 8 years ago, please create a xlat file instead.

Reply:
I've done this using xlat now. This is much nicer

Reply:
Given that linux/lirc.h is bundled, there is no need to test LIRC_* constants,
so please add #unconditional to these xlat files.

Reply:
done